### PR TITLE
Rename GPS to GNSS

### PR DESCRIPTION
**Pull request by JianyuWang0623**
## Summary
* Rename GPS to GNSS.
* The transition from GPS to GNSS primarily stems from two key reasons.
* The primary reason for shifting from GPS to GNSS is that GNSS encompasses all global navigation systems, including GPS, Beidou, Galileo, and others. This broader scope ensures compatibility and flexibility.
* Another reason is to align with Android's conventions and standards. By the way, we are internally integrating Nuttx's GNSS data into Android's GNSS HAL through uORB, which is why we've introduced new data structures such as measurement, clock, and geofence. For a more detailed understanding, you can refer to the code：
    https://cs.android.com/android/platform/superproject/main/+/main:hardware/interfaces/gnss/2.0/default/?hl=zh-cn

## Impact
* GPS is renamed to GNSS.
* GNSS is broader scope of location / navigation services, it ensures better compatibility and flelxibility.
* Better alignment with Android (GNSS HAL / uORB).

## Testing

Please provide testing details and logs.

### CODE REVIEW

**Comment by JianyuWang0623** [comment]
depends on https://github.com/apache/nuttx/pull/13185

**Review by cederom — CHANGES_REQUESTED** [review]
Thank you @JianyuWang0623 :-)

Please update commit messages so they describe in human readable form why change is necessary, what is does and how.

Please also expand the Summary, Impact, and Testing descriptions in the PR.

**Comment by Donny9** [comment]
> Thank you @JianyuWang0623 :-)
> 
> Please update commit messages so they describe in human readable form why change is necessary, what is does and how.
> 
> Please also expand the Summary, Impact, and Testing descriptions in the PR.
> 
> There are much more changes introduced than just "Rename GPS to GNSS".

@cederom Hello,
The transition from GPS to GNSS primarily stems from two key reasons. 
* The primary reason for shifting from GPS to GNSS is that GNSS encompasses all global navigation systems, including GPS, Beidou, Galileo, and others. This broader scope ensures compatibility and flexibility. 
* Another reason is to align with Android's conventions and standards. By the way, we are internally integrating Nuttx's GNSS data into Android's GNSS HAL through uORB, which is why we've introduced new data structures such as measurement, clock, and geofence. For a more detailed understanding, you can refer to the code：
https://cs.android.com/android/platform/superproject/main/+/main:hardware/interfaces/gnss/2.0/default/?hl=zh-cn

File: include/industry/foc/fixed16/foc_feedforward.h [hunk 1]
```diff
@@ -33,6 +33,11 @@
  * Public Type Definition
  ****************************************************************************/
 
+#ifdef CONFIG_DEBUG_UORB
```
Comment (xiaoxiang781216) [inline include/industry/foc/fixed16/foc_feedforward.h:36]: why change this file

Comment (JianyuWang0623) [reply include/industry/foc/fixed16/foc_feedforward.h]: Sorry, no need, a mistake while processing merge conflicts / depends.

**Comment by JianyuWang0623** [comment]
007d0c496a7024d95929f1c342a9360c06493675 and e4ba666bc5992d81645fe6d7a7f30424f068ee75 are signed-off by @Otpvondoiats

**Review by cederom — APPROVED** [review]
Thank you @JianyuWang0623 :-)

This PR depends on https://github.com/apache/nuttx/pull/13185 that I already marked approved :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**